allauth/socialaccount/providers/__init__.py

```diff
@@ -12,9 +12,16 @@ def __init__(self):
 
     def get_list(self, request=None):
         self.load()
+        try:
+            from django.contrib.sites.models import Site
+            site = Site.objects.get_current(request)
+            ids = [sa.provider for sa in site.socialapp_set.all()]
+        except:
```

Comment:
This was only intended so that in case of **any** failure the default functionality would still be performed. Will check what exceptions could be thrown here.